app.ts

```diff
@@ -0,0 +1,411 @@
+						fromID = fromID + this.difference;
+					}
+					let toID = fromID + (maxRecords - 1);
+					if (fromID > count + 1 || toID > count + 1) {
+						toID = count;
+						fromID = toID - maxRecords;
+						this.currentFromID = fromID;
+					}
+					this.currentFromID = fromID;
+					$(".pagination-item").removeClass("active");
+					$(event.target).addClass("active");
+					const self = this;
+					$(".pagination-item").each(function () {
+						let elementID = <string>($(this).attr('value'));
+						let currentPageString = self.currentPage.toString();
+						if (elementID == currentPageString) {
+							$(this).addClass('active');
+						}
+					});
+					return this.showRecords(fromID, toID)
+						.then(() => {
+							$('.pagination-item').prop('disabled', false)
+						});
+				})
+				.catch(error => {
+					throw new Error(`Failed showing the records when clicking on the page button: ${error}`);
+				});
+		})
+
+		/** Gives the next set of page numbers based on the last page on the pagination. */
+		$(".next").on("click", () => {
+			this.paginationStart = this.paginationEnd + 1;
+			this.paginationEnd = this.paginationStart + 9;
+			$(".pagination").empty();
+			this.pageNumbers(this.paginationStart, this.paginationEnd);
+		});
+
+		/** Gives the previous set of pages numbers based on the last page in the pagination. */
+		$(".prev").on("click", () => {
+			this.paginationEnd = this.paginationStart - 1;
+			this.paginationStart = this.paginationEnd - 9;
+			$(".pagination").empty();
+			this.pageNumbers(this.paginationStart, this.paginationEnd);
+		});
+	}
+
+	/** Handles all the functionality related to the search. */
+	initializeSearch(): void {
+		/** Prevents certain characters to be entered in the input field. */
+		$('.search-input').on('keydown', (e) => {
+			if (e.key === 'e' || e.key === 'E' || e.key === '.' || e.key === '+' || e.key === '*' || e.key === '-') {
+				e.preventDefault();
+			}
+			if (e.key === 'Enter') {
+				$('.heading').trigger('click', '.results-box');
+			}
+		});
+
+		/** Takes the number entered in the search field and calculates a range and render that 
+		 * on to the DOM. */
+		$(".search-input").on("input", (event: any) => {
+			event.preventDefault();
+			return this.apiManager.getRecordCount()
+				.then(count => {
+					let inputNumber = this.input();
+					let inputNumberInt = parseInt(inputNumber);
+					if (inputNumber !== '') {
+						const maxRecords = this.recordsPerPage();
+						let end = Math.ceil(inputNumberInt / maxRecords) * (maxRecords);
+						if (end > (count + 1)) {
+							end = count;
+							this.currentToID = end;
+						}
+						let start = (end - (maxRecords - 1));
+						if ((end - maxRecords) === 0) {
+							end = Math.ceil(inputNumberInt / maxRecords) * (maxRecords);
+							start = end - maxRecords;
+						}
+						this.currentPage = Math.floor(end / maxRecords);
+						if (inputNumberInt < 1000000 && inputNumberInt > 0) {
```

Comment:
```suggestion
						if (inputNumberInt < 1000000 && inputNumberInt > 0) {
```
You have your count. You went all the way to fetch it, use it and not 1000000.